TwitchLib.Api.Core/HttpCallHandlers/TwitchHttpClient.cs

```diff
@@ -122,30 +122,42 @@ public async Task<int> RequestReturnResponseCodeAsync(string url, string method,
 
         private void HandleWebException(HttpResponseMessage errorResp)
         {
+            // Show the actual message that came back from the server. Masking it only makes debugging hard.
+            var actualContent = String.Empty;
+
+            try
+            {
+                actualContent = errorResp.Content.ReadAsStringAsync().Result;
+            }
+            catch (Exception ex)
+            {
+                actualContent = $"Couldn't read response from server: {ex.Message}";
+            }
+
             switch (errorResp.StatusCode)
             {
                 case HttpStatusCode.BadRequest:
-                    throw new BadRequestException("Your request failed because either: \n 1. Your ClientID was invalid/not set. \n 2. Your refresh token was invalid. \n 3. You requested a username when the server was expecting a user ID.", errorResp);
+                    throw new BadRequestException(actualContent + "\n\nYour request may have failed because either: \n 1. Your ClientID was invalid/not set. \n 2. Your refresh token was invalid. \n 3. You requested a username when the server was expecting a user ID.", errorResp);
                 case HttpStatusCode.Unauthorized:
                     var authenticateHeader = errorResp.Headers.WwwAuthenticate;
                     if (authenticateHeader == null || authenticateHeader.Count <= 0)
-                        throw new BadScopeException("Your request was blocked due to bad credentials (Do you have the right scope for your access token?).", errorResp);
-                    throw new TokenExpiredException("Your request was blocked due to an expired Token. Please refresh your token and update your API instance settings.", errorResp);
+                        throw new BadScopeException(actualContent + "\n\nYour request was blocked due to bad credentials (Do you have the right scope for your access token?).", errorResp);
+                    throw new TokenExpiredException(actualContent + "\n\nYour request was blocked due to an expired Token. Please refresh your token and update your API instance settings.", errorResp);
                 case HttpStatusCode.NotFound:
-                    throw new BadResourceException("The resource you tried to access was not valid.", errorResp);
+                    throw new BadResourceException(actualContent + "\n\nThe resource you tried to access was not valid.", errorResp);
                 case (HttpStatusCode)429:
-                    errorResp.Headers.TryGetValues("Ratelimit-Reset", out var resetTime);
-                    throw new TooManyRequestsException("You have reached your rate limit. Too many requests were made", resetTime.FirstOrDefault(), errorResp);
+                    errorResp.Headers.TryGetValues(actualContent + "\n\nRatelimit -Reset", out var resetTime);
```

Comment:
You've done a replace all and accidentally caught a header value

Reply:
Whoops! Quite right, fixed now. Thanks!